run_tests.sh

```diff
@@ -16,9 +29,6 @@ echo "Docker logs for server"
 docker logs test_explorer_server
 echo "Trying curl"
 docker run --rm --network="container:$varA" byrnedo/alpine-curl -f http://localhost:8080/
-docker run --rm --network="container:$varA" byrnedo/alpine-curl -f http://localhost:8080/api/blocks/10 
+docker run --rm --network="container:$varA" byrnedo/alpine-curl -f http://localhost:8080/api/blocks/10
 docker run --rm --network="container:$varA" byrnedo/alpine-curl -f http://localhost:8080/api/blocks/10/transactions
-for imagename in test_web3 test_mongo test_explorer_grabber test_explorer_server; do
-    echo "* Trying to stop ${imagename}..."
-    [ -z $(docker ps -a -f "name=$imagename" --format '{{.Names}}') ] || docker rm -f $imagename
-done
+cleanup_containers
```

Comment:
I added the `trap`s because this kept hanging and I had to kill it with an interrupt and clean it up manually. It's good that it will clean it self up beforehand now, but won't it still leave containers running if it's interrupted? Can we move this closing cleanup call into a `trap` up top instead? It should also make the initial call unnecessary, because it will never leave anything running in the first place.

Reply:
Less important, but IMO a per-container call is cleaner (e.g. `trap cleanup_container(test_explorer_grabber) EXIT SIGINT SIGTERM`). That way it only tries to clean what has actually been launched.

Reply:
Another alternative would be to use `docker-compose`.

Reply:
this 'trap' thing was not working properly on my env(I'm running ubuntu 19, so in theory behaviour should be the same but it wasn't), but I'll check how errors/exits could be handled properly

Reply:
How was it failing? Do we just need a #!/bin/bash or something?

Reply:
nope it was not working as it used to
1) that was the first issue https://unix.stackexchange.com/questions/314554/why-do-i-get-an-error-message-when-trying-to-trap-a-sigint-signal
2) It was not run in case of failure (figuring out now)

Reply:
Strange. I thought it was working fine for me on Ubuntu 18.04 and 18.10